### PR TITLE
Add typescript definitions

### PR DESCRIPTION
**Pull request by janslow**
Assists developers using TypeScript so they don't need to write their own definitions when using this package.

### CODE REVIEW

**Comment by evanshortiss** [comment]
@helloIAmPau any interest in merging this? Would be useful. If not then it'd be good to know so @janslow could push these to DefinitelyTyped instead.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+export class Spinner {
+  static setDefaultSpinnerString(spinnerString: string): void;
+  static setDefaultSpinnerString(builtInSpinner: number): void;
+  static setDefaultSpinnerDelay(spinnerDelay: number): void;
+  static setSpinnerTitle(spinnerTitle: string): void;
+
+  constructor(title: string);
```
Comment (evanshortiss) [inline index.d.ts]: I think title is optional. So `constructor(title?: string)`

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export class Spinner {
+  static setDefaultSpinnerString(spinnerString: string): void;
+  static setDefaultSpinnerString(builtInSpinner: number): void;
+  static setDefaultSpinnerDelay(spinnerDelay: number): void;
+  static setSpinnerTitle(spinnerTitle: string): void;
```
Comment (evanshortiss) [inline index.d.ts]: Think this one is not static.

**Comment by evanshortiss** [comment]
@janslow added some comments to the PR. One other thing to note is that some methods return `this` so the types can be like so:

```ts
  export class Spinner {
    constructor (title?: string)

    static setDefaultSpinnerString(spinnerString: string|number): Spinner
    static setDefaultSpinnerDelay(spinnerDelay: number): Spinner

    setSpinnerString(spinnerString: string|number): Spinner
    setSpinnerDelay(spinnerDelay: number): Spinner
    setSpinnerTitle(spinnerTitle: string): Spinner

    isSpinning(): boolean

    start(): Spinner
    stop(): Spinner
  }
```

File: index.d.ts (hunk 1 above)
Comment (janslow) [reply index.d.ts]: Fixed

File: index.d.ts (hunk 2 above)
Comment (janslow) [reply index.d.ts]: Fixed

**Comment by helloIAmPau** [comment]
Hello, sorry for the delay, but I was away from Github for a long time. How is the status of this PR? Are the definition added to DefinitelyTyped or should I accept the PR?

**Comment by evanshortiss** [comment]
@helloIAmPau no problem! Looks like @janslow has added them to DefinitelyTyped

```
$ npm view @types/cli-spinner
{ name: '@types/cli-spinner',
  description: 'TypeScript definitions for cli-spinner',
  'dist-tags': 
   { latest: '0.2.0',
     'ts2.0': '0.2.0',
     'ts2.1': '0.2.0',
     'ts2.2': '0.2.0',
     'ts2.3': '0.2.0',
     'ts2.4': '0.2.0',
     'ts2.5': '0.2.0',
     'ts2.6': '0.2.0',
     'ts2.7': '0.2.0',
     'ts2.8': '0.2.0',
     'ts2.9': '0.2.0',
     'ts3.0': '0.2.0',
     'ts3.1': '0.2.0' },
  versions: [ '0.2.0' ],
  maintainers: [ 'types <ts-npm-types@microsoft.com>' ],
  time: 
   { modified: '2018-07-16T18:30:22.136Z',
     created: '2018-01-02T19:36:05.429Z',
     '0.2.0': '2018-01-02T19:36:05.429Z' },
  repository: 
   { type: 'git',
     url: 'https://www.github.com/DefinitelyTyped/DefinitelyTyped.git' },
  contributors: 
   [ { name: 'Jay Anslow',
       url: 'https://github.com/janslow',
       githubUsername: 'janslow' } ],
  license: 'MIT',
  readmeFilename: '',
  version: '0.2.0',
  main: '',
  scripts: {},
  dependencies: { '@types/node': '*' },
  typesPublisherContentHash: '4d000f08d1e6fd5ac29618e066036648632afca4d247c1fa76ab3397fab5a27e',
  typeScriptVersion: '2.0',
  dist: 
   { integrity: 'sha512-M7+zuU/wPkgG74TZnS4lE/FD6pH4HQg/H7QHPInGfPcx+kfc9dam+yWbye6CAqdpUczdMjvN7NfBf1OUEJnAZA==',
     shasum: '50304d21d9a3623a878fc2a42c9bfa6921c6d236',
     tarball: 'https://registry.npmjs.org/@types/cli-spinner/-/cli-spinner-0.2.0.tgz' },
  directories: {} }
```

**Comment by helloIAmPau** [comment]
Ok, so I can close. Thank you

**Pull request closed without merging**